WordPress/Classes/Models/Blog.m

```diff
@@ -101,18 +99,22 @@ - (void)setGeolocationEnabled:(BOOL)value
 #pragma mark -
 #pragma mark Custom methods
 
-- (NSString *)blavatarUrl
+- (NSString *)icon
 {
-    if (_blavatarUrl == nil) {
-        NSString *hostUrl = [[NSURL URLWithString:self.xmlrpc] host];
-        if (hostUrl == nil) {
-            hostUrl = self.xmlrpc;
-        }
+    [self willAccessValueForKey:@"icon"];
+    NSString *icon = [self primitiveValueForKey:@"icon"];
+    [self didAccessValueForKey:@"icon"];
 
-        _blavatarUrl = hostUrl;
+    if (icon) {
+        return icon;
     }
 
-    return _blavatarUrl;
+    // if the icon is not set we can use the host url to construct it
+    NSString *hostUrl = [[NSURL URLWithString:self.xmlrpc] host];
+    if (hostUrl == nil) {
+        hostUrl = self.xmlrpc;
+    }
+    return hostUrl;
```

Comment:
Do we still need to fallback to the host? And if we don't, can we make `blog.icon` a `NSURL`?